app/models/service_template/copy.rb

```diff
@@ -6,12 +6,24 @@ def template_copy(new_name = "Copy of " + name + Time.zone.now.to_s)
       ActiveRecord::Base.transaction do
         dup.tap do |template|
           template.update_attributes(:name => new_name, :display => false)
-          service_resources.each do |sr|
-            resource = sr.resource.respond_to?(:service_template_resource_copy) ? sr.resource.service_template_resource_copy : sr.resource
-            template.add_resource(resource, sr)
-          end
+          service_resources.each { |service_resource| resource_copy(service_resource, template) }
+          custom_buttons.each { |custom_button| custom_button_copy(custom_button, template) }
```

Comment:
I think this is the bug ... `custom_buttons` are not custom buttons assigned to this service template.

If you want that, you need to use `direct_custom_buttons`.
`custom_buttons` gives you those, *and* every single custom button for any ServiceTemplate .. and you don't want to copy those.

(that should fix https://github.com/ManageIQ/manageiq/issues/18954)